src/python/dxpy/bindings/dxdatabase.py

```diff
@@ -0,0 +1,339 @@
+from threading import Lock
+from multiprocessing import cpu_count
+
+import dxpy
+from . import DXDataObject
+from ..exceptions import DXFileError, DXIncompleteReadsError
+from ..utils import warn
+from ..utils.resolver import object_exists_in_project
+from ..compat import BytesIO, basestring, USING_PYTHON2
+from .. import logger
+
+
+DXFILE_HTTP_THREADS = min(cpu_count(), 8)
+MIN_BUFFER_SIZE = 1024*1024
+DEFAULT_BUFFER_SIZE = 1024*1024*16
+if dxpy.JOB_ID:
+    # Increase HTTP request buffer size when we are running within the
+    # platform.
+    DEFAULT_BUFFER_SIZE = 1024*1024*96
+
+MD5_READ_CHUNK_SIZE = 1024*1024*4
+FILE_REQUEST_TIMEOUT = 60
+
+
+def _validate_headers(headers):
+    for key, value in headers.items():
+        if not isinstance(key, basestring):
+            raise ValueError("Expected key %r of headers to be a string" % (key,))
+        if not isinstance(value, basestring):
+            raise ValueError("Expected value %r of headers (associated with key %r) to be a string"
+                             % (value, key))
+    return headers
+
+
+def _readable_part_size(num_bytes):
+    "Returns the file size in readable form."
+    B = num_bytes
+    KB = float(1024)
+    MB = float(KB * 1024)
+    GB = float(MB * 1024)
+    TB = float(GB * 1024)
+
+    if B < KB:
+        return '{0} {1}'.format(B, 'bytes' if B != 1 else 'byte')
+    elif KB <= B < MB:
+        return '{0:.2f} KiB'.format(B/KB)
+    elif MB <= B < GB:
+        return '{0:.2f} MiB'.format(B/MB)
+    elif GB <= B < TB:
+        return '{0:.2f} GiB'.format(B/GB)
+    elif TB <= B:
+        return '{0:.2f} TiB'.format(B/TB)
+
+def do_debug(msg):
+    logger.info(msg)
+
+class DXDatabase(DXDataObject):
+    '''Remote database object handler.
+
+    :param dxid: Object ID
+    :type dxid: string
+    :param project: Project ID
+    :type project: string
+    :param mode: One of "r", "w", or "a" for read, write, and append modes, respectively.
+                 Use "b" for binary mode. For example, "rb" means open a file for reading
+                 in binary mode.
+    :type mode: string
+
+    .. note:: The attribute values below are current as of the last time
+              :meth:`~dxpy.bindings.DXDataObject.describe` was run.
+              (Access to any of the below attributes causes
+              :meth:`~dxpy.bindings.DXDataObject.describe` to be called
+              if it has never been called before.)
+
+    .. py:attribute:: media
+
+       String containing the Internet Media Type (also known as MIME type
+       or Content-type) of the file.
+
+    .. automethod:: _new
+
+    '''
+
+    _class = "database"
+
+    _describe = staticmethod(dxpy.api.file_describe)
+
+    _http_threadpool_size = DXFILE_HTTP_THREADS
+    _http_threadpool = dxpy.utils.get_futures_threadpool(max_workers=_http_threadpool_size)
+
+    NO_PROJECT_HINT = 'NO_PROJECT_HINT'
+
+    def __init__(self, dxid=None, project=None, mode=None, read_buffer_size=DEFAULT_BUFFER_SIZE,
```

Comment:
Well, it's intended to represent database data object. However, it's use cases are limited to downloading files (dx get) for now, so that representation is a little vague I admit. As for action items in the immediate term, @durae , I'll just remove any remaining unnecessary items like the ones you named, and I'll also refrain from adding stuff (e.g. databaseName) that would make sense but which aren't currently needed. Reasonable?

Reply:
yes, sounds alright for now. We can add it there later when needed for other database specific commands